File: hw9/HW9.java
package hw9;

import java.util.Map;
import java.util.HashMap;

public final class HW9 {
    private HW9() {
        //
    }

    public static int romanToInteger(String str) {
        int intConversion = 0;

        Map<Character, Integer> romanSymbol = new HashMap<>();
        romanSymbol.put('I', 1);
        romanSymbol.put('V', 5);
        romanSymbol.put('X', 10);
        romanSymbol.put('L', 50);
        romanSymbol.put('C', 100);
        romanSymbol.put('D', 500);
        romanSymbol.put('M', 1000);

        for (int i = 0; i < str.length() - 1; i++) {
            if (romanSymbol.get(str.charAt(i)) >= romanSymbol.get(str.charAt(i + 1))) 
                intConversion += romanSymbol.get(str.charAt(i));
            else intConversion -= romanSymbol.get(str.charAt(i));
        }

        return intConversion;
    }
}

File: hw9/hw9.js
function romanToInteger(str) {
    let intConversion = 0;
    let romanSymbol = {
        'I': 1,
        'V': 5,
        'X': 10,
        'L': 50,
        'C': 100,
        'D': 500,
        'M': 1000
    };

    for (let i = 0; i < str.length - 1; i++) {
        if (romanSymbol[str[i]] >= romanSymbol[str[i+1]]) intConversion += romanSymbol[str[i]];
        else intConversion -= romanSymbol[str[i]];
    }

    intConversion += romanSymbol[str[str.length - 1]];

    return intConversion
}



